### PR TITLE
Make domain tag configurable

### PR DESCRIPTION
**Pull request by vadorovsky**
The authors of the Poseidon hash recommend using domain separation when an application utilizes multiple instances of the Poseidon hash (section 4.2 of Poseidon hash paper[0]). Before this change, we were hard coding the domain tag to zero. Let's make it configurable.

Fixes: #18

### CODE REVIEW

**Comment by ananas-block** [comment]
Could we put the domain tag into the parameters?
(I think it’s going to be cumbersome to always specify none in every function call.)

File: light-poseidon/tests/bn254_fq_x5.rs [hunk 1]
```diff
@@ -19,6 +19,22 @@ fn test_poseidon_bn254_x5_fq_input_ones_twos() {
...
+    let hash = hasher.hash(&[input1, input2]).unwrap();
+
+    assert_eq!(
+        hash.into_bigint().to_bytes_be(),
+        [
+            13, 84, 225, 147, 143, 138, 140, 28, 125, 235, 94, 3, 85, 242, 99, 25, 32, 123, 132,
+            254, 156, 162, 206, 27, 38, 231, 53, 200, 41, 130, 25, 144
+        ]
+    );
+}
```
Comment (ananas-block) [inline light-poseidon/tests/bn254_fq_x5.rs:36]: we should add another test that the result with domain tag Fr::one() is not equal to the result with domain tag Fr::zero()

**Review by ananas-block — CHANGES_REQUESTED** [review]

File: light-poseidon/tests/bn254_fq_x5.rs (hunk 1 above)
Comment (vadorovsky) [reply light-poseidon/tests/bn254_fq_x5.rs:40]: Done

Comment (ananas-block) [reply light-poseidon/tests/bn254_fq_x5.rs:40]: thanks

**Review by ananas-block — APPROVED** [review]
